hack/multikueue/setup-multikueue.sh

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The Kubernetes Authors.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# =============================================================================
+# CONFIGURATION AND SETUP
+# =============================================================================
+
+SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
```

Comment:
Actually, we already have the makefile goal `test-multikueue-e2e` which provisions the setup. 

Instead of duplicating I would like to reuse the goal, just parametrize it. I think what we just need is ginkgo focus which skips any tests, and some option to SKIP_CLUSTER_CLEANUP, so that the clusters are left after "tests".
Maybe focus could also be an overridable parameter, so that you can run just one test, when skip cluster cleanup for manual inspection.

Wdyt?

Reply:
oh thats a good idea. Yeah I was going to otherwise pull out common functions in some sort of common-multikueue file in bash.

Reply:
Will pick up this PR down the line! I think your suggestion via the ginkgo focus is feasible. I just need to wrap some tests in a context to separate from the setup stuff. So `SKIP_CLUSTER_CLEANUP` can be some make command that calls the correct focus